lib/gltf.js

```diff
@@ -154,185 +43,301 @@ function createGltf(data, inputPath, modelName) {
+    function getTextureId(imagePath) {
+        if (!defined(imagePath) || !defined(images[imagePath])) {
+            return undefined;
+        }
+        return 'texture_' + getImageId(imagePath);
     }
 
-    gltf.buffers[bufferId] = {
-        byteLength : bufferByteLength,
-        type : 'arraybuffer',
-        uri : bufferUri
-    };
+    function createMaterial(material, hasNormals) {
+        var ambient = defaultValue(defaultValue(getTextureId(material.ambientColorMap), material.ambientColor), [0, 0, 0, 1]);
+        var diffuse = defaultValue(defaultValue(getTextureId(material.diffuseColorMap), material.diffuseColor), [0.5, 0.5, 0.5, 1]);
+        var emission = defaultValue(defaultValue(getTextureId(material.emissionColorMap), material.emissionColor), [0, 0, 0, 1]);
+        var specular = defaultValue(defaultValue(getTextureId(material.specularColorMap), material.specularColor), [0, 0, 0, 1]);
+        var alpha = defaultValue(defaultValue(material.alpha), 1.0);
+        var shininess = defaultValue(material.specularShininess, 0.0);
+        var hasSpecular = (shininess > 0.0) && (specular[0] > 0.0 || specular[1] > 0.0 || specular[2] > 0.0);
+
+        var transparent;
+        var transparency = 1.0;
+        if (typeof diffuse === 'string') {
+            transparency = alpha;
+            transparent = images[material.diffuseColorMap].transparent || (transparency < 1.0);
+        } else {
+            diffuse[3] = alpha;
+            transparent = diffuse[3] < 1.0;
+        }
 
-    gltf.bufferViews[bufferViewVertexId] = {
-        buffer : bufferId,
-        byteLength : vertexArrayByteLength,
-        byteOffset : 0,
-        target : WebGLConstants.ARRAY_BUFFER
-    };
-    gltf.bufferViews[bufferViewIndexId] = {
-        buffer : bufferId,
-        byteLength : indexArrayByteLength,
-        byteOffset : vertexArrayByteLength,
-        target : WebGLConstants.ELEMENT_ARRAY_BUFFER
-    };
+        var doubleSided = transparent;
 
-    for (i = 0; i < primitivesLength; ++i) {
-        var primitive = primitives[i];
-        gltf.accessors[getAccessorIndexId(i)] = {
-            bufferView : bufferViewIndexId,
-            byteOffset : primitive.indexOffset * indexComponentSize,
-            byteStride : 0,
-            componentType : indexComponentType,
-            count : primitive.indexCount,
-            type : 'SCALAR'
+        if (!hasNormals) {
+            // Constant technique only factors in ambient and emission sources - set emission to diffuse
+            emission = diffuse;
+            diffuse = [0, 0, 0, 1];
+        }
+
+        // It's not completely clear whether transparent and doubleSided belong under values or KHR_materials_common
+        // Put under both for now to handle both situations.
+        // https://github.com/KhronosGroup/glTF/tree/master/extensions/Khronos/KHR_materials_common
+        // https://github.com/KhronosGroup/glTF/issues/632
```

Comment:
You should write an `obj2gltf` issue for this as well, so that this comment doesn't just sit here forever, forgotten.